pyanaconda/modules/subscription/subscription.py

```diff
@@ -572,14 +572,29 @@ def get_rhsm_config_defaults(self):
         or else the method might return non-default (Anaconda overwritten)
         data.
 
+        NOTE: While RHSM GetAll() DBus call returns a nested dictionary,
+              we turn it into a flat key/value dict, in the same format SetAll()
+              uses.
+
         :return : dictionary of default RHSM configuration values
         :rtype: dict
         """
         if self._rhsm_config_defaults is None:
             # config defaults cache not yet populated, do it now
             proxy = self.rhsm_observer.get_proxy(RHSM_CONFIG)
             # turn the variant into a dict with get_native()
-            self._rhsm_config_defaults = get_native(proxy.GetAll(""))
+            nested_dict = get_native(proxy.GetAll(""))
+            # RHSM returns a nested dict with categories on top
+            # and category keys & values inside. This is not convenient
+            # for setting keys based on original values, so
+            # let's normalize the dict to the flat key based
+            # structure similar to what's used by SetAll().
+            flat_dict = {}
```

Comment:
I would prefer to keep the logic in a function that could be somewhere else.

Reply:
Moved to a helper function. :)